### ISSUE TITLE
AV1505: False positive on top-level programs

### ISSUE THREAD

**Issue by bkoelman** [bkoelman/CSharpGuidelinesAnalyzer#126]
#### Analyzer
NamespaceShouldMatchAssemblyNameAnalyzer (AV1505)

#### Repro steps
Create a top-level program (without declaring the Program class). For example:
```c#
System.Console.WriteLine("");
```

#### Expected behavior
No warning, because the user did not declare a Program class, nor a namespace. The analyzer should exclude top-level programs (where the Program class is generated by the compiler).

#### Actual behavior
A warning is reported that class "Program" is declared in the global namespace. Double-clicking on the warning jumps to the first statement in the file, which is unrelated to the problem.

**Comment by bkoelman** [comment]
https://github.com/dotnet/roslyn/issues/58120
